src/idl_gen_pybind.cpp

```diff
@@ -0,0 +1,1259 @@
+  }
+
+  // Generates an internal property that marks that this is a pybind flatbuffer
+  // class.
+  // TODO(michael-ahn): This is better as a class property once supported by
+  // pybind.
+  void GenerateInternalPybindTypeMarker(int base_type) {
+    code_ += Indent() +
+             "{{BIND_VAR}}.def_property_readonly(\"_fbs_pybind_type\", "
+             "[](py::handle) -> int { return " +
+             NumToString(base_type) + "; });";
+  }
+
+  // Generates a pybind definition for a struct.
+  void GenerateStruct(const StructDef &def) {
+    SetCodeValuesStruct(def);
+    GenerateComment(def.doc_comment);
+    GenerateInternalPybindTypeMarker(BASE_TYPE_STRUCT);
+
+    // Default constructor.
+    code_ += Indent() + "{{BIND_VAR}}.def(py::init<>());";
+
+    auto field_defs = GetFieldDefs(def);
+
+    // Keyword args constructor.
+    if (field_defs.size() > 0) {
+      std::vector<std::string> arg_types, py_args;
+      arg_types.reserve(field_defs.size());
+      py_args.reserve(field_defs.size());
+      for (const auto *field : field_defs) {
+        arg_types.push_back(CppArgumentType(field->value.type));
+        py_args.push_back("py::arg(\"" + field->name +
+                          "\") = " + PyArgDefaultValue(*field));
+      }
+      code_ += Indent() + "{{BIND_VAR}}.def(";
+      code_ += Indent(3) + "py::init<" + StrJoin(arg_types, ", ") + ">(),";
+      code_ += Indent(3) + StrJoin(py_args, ", ") + ");";
+    }
+
+    // Generate accessor bindings.
+    for (const auto *field : field_defs) {
+      const auto &field_type = field->value.type;
+      code_.SetValue("CPP_FIELD", cpp_namer_.Field(*field));
+      code_.SetValue("PY_FIELD", py_namer_.Field(*field));
+
+      if (IsStruct(field_type)) {
+        code_ += Indent() + "{{BIND_VAR}}.def_property(";
+        code_ += Indent(3) + "\"{{PY_FIELD}}\",";
+        code_ += Indent(3) + "&{{CPP_TYPE}}::{{CPP_FIELD}},";
+        code_ += Indent(3) + "[]({{CPP_TYPE}} &self, " +
+                 CppArgumentType(field_type) + " value) {";
+        code_ += Indent(4) + "self.mutable_{{CPP_FIELD}}() = value;";
+        code_ += Indent(3) + "});";
+        continue;
+      }
+
+      if (IsArray(field_type)) {
+        code_ += Indent() +
+                 "{{BIND_VAR}}.def_property_readonly(\"{{PY_FIELD}}\", "
+                 "&{{CPP_TYPE}}::mutable_{{CPP_FIELD}});";
+        continue;
+      }
+
+      // POD types.
+      if (opts_.mutable_buffer) {
+        code_ += Indent() +
+                 "{{BIND_VAR}}.def_property(\"{{PY_FIELD}}\", "
+                 "&{{CPP_TYPE}}::{{CPP_FIELD}}, "
+                 "&{{CPP_TYPE}}::mutate_{{CPP_FIELD}});";
+      } else {
+        code_ += Indent() +
+                 "{{BIND_VAR}}.def_property_readonly(\"{{PY_FIELD}}\", "
+                 "&{{CPP_TYPE}}::{{CPP_FIELD}});";
+      }
+    }
+
+    // __repr__
+    GenerateReprBinding(field_defs);
+
+    code_ += "";
+  }
+
+  void GenerateTable(const StructDef &def) {
```

Comment:
Oh man, it is so much nicer having these all inlined like this instead of the other generator files where finding out what made a given text output in the generated file involves jumping around into 100 different little functions.